pkg/harvester/edit/harvesterhci.io.volume.vue

```diff
@@ -224,25 +224,29 @@ export default {
 
     rebuildStatus() {
       return this.value.longhornEngine?.status?.rebuildStatus;
-    },
-
-    isLonghornV2() {
-      return this.value.storageClass?.isLonghornV2;
     }
   },
 
   methods: {
+    getAccessMode() {
+      const storageClassName = this.value.spec.storageClassName;
+      const storageClass = this.storageClasses.find(sc => sc.name === storageClassName);
+
+      let readWriteOnce = this.value.isLvm || this.value.isLonghornV2;
+
+      if (storageClass) {
```

Comment:
In create mode, the volume `provisioner` & `dataEngine` must be the ones from the Storage Class, as the user can change the Storage Class value.